### PR TITLE
[feat] Support different target platform module types for java modules

### PR DESCRIPTION
**Pull request by tal-sapan**
## Description
Add option to map module type to different target platform module types
according to module parameter/property in a target platform. The mapping
that specifies more details (parameters or properties) is taken.

For java modules, support the java.tomee and com.sap.java module types
according to the relevant parameters/properties in each target platform.

### Checklist
- [x] Code compiles correctly
- [x] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [x] Formating and linting run locally successfully
- [x] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: internal/platform/process.go [hunk 1]
```diff
@@ -21,14 +21,41 @@ func Unmarshal(data []byte) (Platforms, error) {
...
 	}
 }
 
+// Satisfies checks if the module m satisfies the conditions defined in the configuration mc.
+//
+// If it doesn't satisfy the conditions, ok will be false and accuracy will be less than 0.
+//
+// If it satisfies the conditions, accuracy will be higher the more conditions there are inside the configuration
+// (in other words, the more specific match is considered more accurate).
+func satisfies(m *mta.Module, mc *Properties) (ok bool, accuracy int) {
```
Comment (ShimiT) [inline internal/platform/process.go]: IMO I think it's better to give it a name like "satisfiesModuleCfg"

File: configs/platform_cfg.yaml [hunk 2]
```diff
@@ -26,6 +26,11 @@ platform:
   - native-type: hdb
```
Comment (ShimiT) [inline configs/platform_cfg.yaml]: 1. you are missing the binary generation
2. For java module, do you plan to add integration test?

File: internal/platform/process.go (hunk 1 above)
Comment (tal-sapan) [reply internal/platform/process.go]: I'll rename

File: configs/platform_cfg.yaml (hunk 2 above)
Comment (tal-sapan) [reply configs/platform_cfg.yaml]: 1. No, it's in the change (internal/platform/platform_cfg.go)
2. I tested it manually, but I didn't see any tests for other module types (that's why I didn't add tests for the new npm module types I added either). Where should I add it?

**Pull request merged**